esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -1056,44 +1056,18 @@ unsigned WsWuInfo::getWorkunitThorLogInfo(IArrayOf<IEspECLHelpFile>& helpers, IE
     IArrayOf<IConstThorLogInfo> thorLogList;
     if (cw->getWuidVersion() > 0)
     {
-        StringAttr clusterName(cw->queryClusterName());
-        if (!clusterName.length()) //Cluster name may not be set yet
-            return countThorLog;
-
-        Owned<IConstWUClusterInfo> clusterInfo = getTargetClusterInfo(clusterName.str());
-        if (!clusterInfo)
-        {
-            IWARNLOG("Cannot find TargetClusterInfo for workunit %s", cw->queryWuid());
-            return countThorLog;
-        }
-
-        unsigned numberOfSlaveLogs = clusterInfo->getNumberOfSlaveLogs();
+        IArrayOf<IConstWUThorLogInfo> wuThorLogs;
+        if (version < 1.78)
+            cw->getWUThorLogInfo(wuThorLogs);
+        else
+            cw->getWUThorLogInfoLW(wuThorLogs);
```

Comment:
The code here is to call getWUThorLogInfo() for old ESP client version (< 1.78). The getWUThorLogInfo() returns groupName and logDate. The getWUThorLogInfoLW() does not. ESP tries not to change old behaviour in case of breaking old ESP client. 

I may encapsulate the logic into one IConstWorkUnit call, like getWUThorLogInfo(bool returnGroupName, bool returnLogDate, ...) . Please comment.